### ISSUE TITLE
Superset can't access non-development environments

### ISSUE THREAD

**Issue by kainchow** [apache/superset#28707]
### Bug description

When I use `docker compose -f docker-compose-non-dev.yml up` to start superset, it enters the development environment, and using docker compose up also enters the development environment, not the production environment.

### How to reproduce the bug

1. `git clone https://github.com/apache/superset.git`
2. Go to the superset directory
3. Execute the command: `docker compose -f docker-compose-non-dev.yml up`

### Screenshots/recordings

![111](https://github.com/apache/superset/assets/28430952/31027df0-3158-45e8-a621-200c4add84be)

### Superset version

master / latest-dev

### Python version

3.9

### Node version

16

### Browser

Chrome

### Additional context

Without making any project modifications, the command is executed directly after fetching the project from github

### Checklist

- [X] I have searched Superset docs and Slack and didn't find a solution to my problem.
- [X] I have searched the GitHub issue tracker and didn't find a similar bug report.
- [X] I have checked Superset's logs for errors and if I found a relevant Python stacktrace, I included it here as text in the "additional context" section.

**Comment by kainchow** [comment]
I created a new superset_config_docker.py file in the docker/pythonpath_dev directory and added SECRET_KEY, but it still enters the development environment when executing `docker compose -f docker-compose-non-dev up`, here is superset_config_docker.py: 
```python
import os
from superset.utils import core as utils

FEATURE_FLAGS = {
    "ENABLE_TEMPLATE_PROCESSING": True,
}

ENABLE_PROXY_FIX = True
SECRET_KEY = "jwoxth9akQkkzTnen2P4BRQgTRoT8fsv8f3XTKalqE2Ur2ZHQ1NeZxdVxU70Uake"

BABEL_DEFAULT_LOCALE='zh'
BABEL_DEFAULT_FOLDER = 'superset/translations'
LANGUAGES = {
    'zh': {'flag': 'cn', 'name': '简体中文'},
    'en': {'flag': 'us', 'name': 'English'}
}

WTF_CSRF_ENABLED = False
TALISMAN_ENABLED = utils.cast_to_boolean(os.environ.get("TALISMAN_ENABLED", False))

# Maximum number of rows returned for any analytical database query
SQL_MAX_ROW = 10000000
# default row limit when requesting chart data
# ROW_LIMIT = 10000000
FAVICONS = [{"href": "/static/assets/images/superset-mz-black.png"}]
```

**Comment by kainchow** [comment]
By the way, BABEL_DEFAULT_LOCALE and LANGUAGES are effective.
![image](https://github.com/apache/superset/assets/28430952/c03eadbb-c332-44cd-9732-29f5df13efe4)

**Comment by kainchow** [comment]
When I use tag=3.1.3, superset works fine for production, but the latest version, 4.0.1, doesn't work!
![image](https://github.com/apache/superset/assets/28430952/4653bf92-6d10-4d03-8064-768b9d573676)

**Comment by kainchow** [comment]
4.0.1 worked for me, but 4.0.1rc not

**Comment by matteomarjanovic** [comment]
@kainchow it doesn't seem to be solved with the new release (4.1.0), right?
